### PR TITLE
Mobilenav - Refactor

### PR DESCRIPTION
**Pull request by DarrachBarneveld**
# ✨ Codu Pull Request 💻

Codu Logo

## Pull Request details:

- Refactored Mobile Nav into separate component
- Added types.ts file into types folder
- Added slight animation to dropdown menu and hamburger icon
- Added simplified mobile theme toggle switch

## Any Breaking changes:
None

## Associated Screenshots:
    
Old behaviour 
prev-mobile

New behaviour
new-mobile

--

Please note all commented code to be removed pending approval on refactor. 

In association with issue #63 and #464

### CODE REVIEW

**Comment by NiallJoeMaher** [comment]
@DarrachBarneveld would you open up issues for the menu button and the notification icon? Would be cool to get these in light/dark mode too! Reviewing this code now

File: components/Nav/Nav.tsx [hunk 1]
```diff
@@ -213,18 +217,19 @@ const Nav = () => {
                 </Link>
                 {/* Mobile menu button */}
                 <Disclosure.Button className="inline-flex items-center justify-center p-2 rounded-md text-neutral-400 hover:text-white hover:bg-neutral-900 focus:outline-none focus:ring-2 focus:ring-inset focus:ring-white">
-                  <span className="sr-only">Open main menu</span>
+                  {/* <span className="sr-only">Open main menu</span>
```
Comment (NiallJoeMaher) [inline components/Nav/Nav.tsx]: Do we not need this anymore?

File: components/Nav/Nav.tsx [hunk 2]
```diff
@@ -213,18 +217,19 @@ const Nav = () => {
...
                   {open ? (
```
Comment (NiallJoeMaher) [inline components/Nav/Nav.tsx]: Usual comment of look over for any commented code and throw it away 😉

**Review by NiallJoeMaher — CHANGES_REQUESTED** [review]

**Comment by NiallJoeMaher** [comment]
The build failed for some reason! Check your build locally to make sure there are no issues. 🦾

File: components/Nav/Nav.tsx (hunk 2 above)
Comment (DarrachBarneveld) [reply components/Nav/Nav.tsx]: I will remove all commented code now and checkout the build :)

**Comment by NiallJoeMaher** [comment]
Build is still failing sadly 😢 `ci-build` is the command that runs on Vercel.

File: components/Nav/MobileNav.tsx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+  return (
+    <Transition
+      enter="transition-transform duration-200"
+      enterFrom="transform -translate-y-full opacity-0"
+      enterTo="transform translate-y-0 opacity-100"
+      className="absolute w-screen z-10 bg-neutral-100 dark:bg-black"
+    >
+      <Disclosure.Panel className="md:hidden relative z-10 border-b-2 border-black dark:border-white">
+        <div className="px-2 pt-2 pb-3 space-y-1">
+          {navigation.map((item) =>
```
Comment (xiaoniuniu89) [inline components/Nav/MobileNav.tsx]: great job! I am wondering if we can break some of these nested ternaries into separate components for readability? 

Something like

`<NavItem item={item} />`

Where NavItem is a new component that handles the different types of navigation items. This way, the main component stays a bit cleaner

File: components/Nav/MobileNav.tsx [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+                  {item.name}
+                </Disclosure.Button>
+              </Link>
+            )
+          )}
+
+          <div className="pt-3 pb-3 border-t border-neutral-700 flex flex-col space-y-1">
+            {session ? (
+              <>
+                {userSubNav.map((item) => (
```
Comment (xiaoniuniu89) [inline components/Nav/MobileNav.tsx]: same here: 

something like 
`<SubNav session={session} />`

**Review by xiaoniuniu89 — APPROVED** [review]

File: components/Nav/MobileNav.tsx (hunk 4 above)
Comment (DarrachBarneveld) [reply components/Nav/MobileNav.tsx]: I can further refactor as you please. Will begin tomorrow 👍

**Review by NiallJoeMaher — APPROVED** [review]

**Comment by NiallJoeMaher** [comment]
Build is failing but looks great. Good job!

**Pull request merged**